Add support for last name prior to first for guest search

Timo614 opened the pull request:
### References Issues:
References: #1283

### Description:
This adds support for searching by last name prior to first. It wasn't in the original acceptance criteria for the parent issue https://github.com/big-neon/bigneon/issues/165 but may have value in countries where last name generally is written prior to first like Japan as it may feel more natural to search that way. I also ignore the commas in the search query in case people try to emulate the names written below now that we support last name first.

### Examples
Searching normally:
```
curl "http://localhost:9000/events/5e1bb539-ec9e-48b2-bba1-96a8ddb6ed63/guests?query=tay%20l" -i --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIzYzY4MDk3YS0zOTU4LTQ4ODAtYWQ0Yy1hYThkMGU5ZTMyMDMiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTU5NTY0MDUwfQ.-VmQ9B3HKM7sCjoBnofIDjPPL-Ee4hDMaD-utENkr-I"

{"data":[{"id":"72f8d294-a34c-4a2d-aab7-29073665f3ef","ticket_type":"Default_Pricing_1559326497","user_id":"6c92bed1-0150-449d-8380-8d4d4270ab3d","order_id":"279f5626-170e-4313-8c3b-7f2b298504c0","order_item_id":"d4940962-b8fc-4b75-a998-2902ace4a35a","price_in_cents":2510,"first_name":"Tayo","last_name":"Little","email":"user1559326587600414623@tari.com","phone":"555","redeem_key":"FCI5U5U4K","redeem_date":null,"status":"Purchased","event_id":"5e1bb539-ec9e-48b2-bba1-96a8ddb6ed63","event_name":"Event Helper","door_time":"2020-11-13T11:00:00","event_start":"2020-11-13T12:00:00","venue_id":"1465bdfc-9e2d-4265-a0ed-aa300a07c658","venue_name":"Test venue_1559326497","updated_at":"2019-05-31T18:17:18.860745","redeemed_by":null,"redeemed_at":null,"refund_supported":true}],"paging":{"page":0,"limit":100,"sort":"","dir":"Asc","total":0,"tags":{"changes_since":null,"query":"tay l"}}}
```

First name prior to last name:
```
curl "http://localhost:9000/events/5e1bb539-ec9e-48b2-bba1-96a8ddb6ed63/guests?query=li%20ta" -i --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIzYzY4MDk3YS0zOTU4LTQ4ODAtYWQ0Yy1hYThkMGU5ZTMyMDMiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTU5NTY0MDUwfQ.-VmQ9B3HKM7sCjoBnofIDjPPL-Ee4hDMaD-utENkr-I"

{"data":[{"id":"72f8d294-a34c-4a2d-aab7-29073665f3ef","ticket_type":"Default_Pricing_1559326497","user_id":"6c92bed1-0150-449d-8380-8d4d4270ab3d","order_id":"279f5626-170e-4313-8c3b-7f2b298504c0","order_item_id":"d4940962-b8fc-4b75-a998-2902ace4a35a","price_in_cents":2510,"first_name":"Tayo","last_name":"Little","email":"user1559326587600414623@tari.com","phone":"555","redeem_key":"FCI5U5U4K","redeem_date":null,"status":"Purchased","event_id":"5e1bb539-ec9e-48b2-bba1-96a8ddb6ed63","event_name":"Event Helper","door_time":"2020-11-13T11:00:00","event_start":"2020-11-13T12:00:00","venue_id":"1465bdfc-9e2d-4265-a0ed-aa300a07c658","venue_name":"Test venue_1559326497","updated_at":"2019-05-31T18:17:18.860745","redeemed_by":null,"redeemed_at":null,"refund_supported":true}],"paging":{"page":0,"limit":100,"sort":"","dir":"Asc","total":0,"tags":{"changes_since":null,"query":"li ta"}}}
```

Last name with a comma followed by first name:
```
curl "http://localhost:9000/events/5e1bb539-ec9e-48b2-bba1-96a8ddb6ed63/guests?query=little,%20ta" -i --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIzYzY4MDk3YS0zOTU4LTQ4ODAtYWQ0Yy1hYThkMGU5ZTMyMDMiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTU5NTY0MDUwfQ.-VmQ9B3HKM7sCjoBnofIDjPPL-Ee4hDMaD-utENkr-I"

{"data":[{"id":"72f8d294-a34c-4a2d-aab7-29073665f3ef","ticket_type":"Default_Pricing_1559326497","user_id":"6c92bed1-0150-449d-8380-8d4d4270ab3d","order_id":"279f5626-170e-4313-8c3b-7f2b298504c0","order_item_id":"d4940962-b8fc-4b75-a998-2902ace4a35a","price_in_cents":2510,"first_name":"Tayo","last_name":"Little","email":"user1559326587600414623@tari.com","phone":"555","redeem_key":"FCI5U5U4K","redeem_date":null,"status":"Purchased","event_id":"5e1bb539-ec9e-48b2-bba1-96a8ddb6ed63","event_name":"Event Helper","door_time":"2020-11-13T11:00:00","event_start":"2020-11-13T12:00:00","venue_id":"1465bdfc-9e2d-4265-a0ed-aa300a07c658","venue_name":"Test venue_1559326497","updated_at":"2019-05-31T18:17:18.860745","redeemed_by":null,"redeemed_at":null,"refund_supported":true}],"paging":{"page":0,"limit":100,"sort":"","dir":"Asc","total":0,"tags":{"changes_since":null,"query":"little, ta"}}}
```

## Release Details:
### Migrations
 * No Migrations

### Environment Variables
 * No change

Krakaw commented on db/src/models/events.rs:
```diff
@@ -1101,7 +1101,11 @@ impl Event {
-            .bind::<Nullable<Text>, _>(query.map(|q| str::replace(&q, " ", "%")))
+            .bind::<Nullable<Text>, _>(
+                query
+                    .map(|q| str::replace(&q, " ", "%"))
```
I'm not sure how much of a performance knock we would take, but what do you think about doing a:
```suggestion
                    .map(|q| str::replace(&q, " ", "").split("").collect::<Vec<&str>>().join("%"))
```

Timo614 replied:
I doubt it'd use an index regardless since we're wrapping the query in wildcards anyhow so probably not a huge deal performance wise regardless. That's to allow no spaces between the first and last name?

Timo614 replied:
Didn't break the tests and seems fine so just switched to that approach. Gives us more flexibility with the search at least.

Timo614 commented:
Sounds good, I moved it out of the query so it's a bit easier to parse (also just removed the commas versus adding a wildcard since the new logic splits on each character anyhow). Will merge once the tests pass.
